### PR TITLE
Redesigns and Expands Permabrig 

### PR DESCRIPTION
**Pull request by Bawhoppen**
:cl:
add: Permabrig redesigned for Boxstation. (Other maps hopefully coming in time...)
add: Prisoners now may start with some contraband hidden around on Box.
add: Box Permabrig now has a license plate pressing machine, which allows raw license plates (from cargo), to be exported for a profit. Of course, this requires some effort on the prisoner's part.
/:cl:

This PR makes Perma into a miniature caricature of a real prison, with some stereotypical rooms you might find in one. This is just a start on Box, although I would like to do the same for all maps, based roughly on a similar layout. Image of ~~first~~ second draft for Box: 
image

Rooms include: Cell Block (w/ 8 cells), Showers, Cafeteria, Visitation, Transfer Area, Isolation Cell, Laundry, Cleaning Closet, Yard, Forestry, Workshop, and Common Room. 
Should be impossible for prisoners to escape roundstart (hopefully)- Circumstances need to change through outside player involvement, or natural occurences

Minor features added:
License plate press, requires prisoners to insert blank license plate, activate it, wait a couple seconds, pull it out, repeat. Code coming from Floyd in a little bit.
Contraband lootdrop spawners placed around in a few areas.
Laundry cart.
Serving dishes for prison cafeteria, alongside new food reagent to go along with it. They're just re-purposed reagent dispensers that can be used with bowls.
Cafeteria trays. Just normal trays with different look.

Sprites coming for all these things soon.

This PR is unrelated to suggested changes for removing timer-based cells. This is only focused on the permabrig itself.

**WIP.** Has 150 hours of testing, so expect many bugs.

General philosophy behind design: While this perma is certainly larger, I do think I avoided making it too large to be ridiculous. Even now, it's still a generally cramped and small area. It now also has divisions and separate areas out of sight from each other. This is such a big part of the normal game on the station, I thought it would serve well to be in perma as well, although in a toned down sense. 
This principle allows you to be doing something in one area, while unaware of the events happening elsewhere. An example of this may be someone assassinating a rival gang member in the showers while you're in the kitchen. This adds to the paranoia, and allows for more intriguing storylines to play out.
Having it be the size it is, also allows for prisoners to potentially hide contraband, or plot out escape plans (although this requires actually planning and doing things to achieve it). Before this, when the warden and officers came in, they instantly had the place under lockdown, or were immediately overpowered by the prisoners. This will now lead to a more interesting dynamic hopefully. Relatedly, I have removed any sort of lockdowns or blastdoors from perma, even though they would make sense to have. This is an element of fallibility, something which is key to the spirit of ss13, and something which should hopefully make for far more interesting scenarios. 

Overall, I want this new layout to allow for a range of fun possibilities, such as escaping by hiding in a crate of finished license-plates bound for cargo, smuggling in contraband through visitation, or simply having honest jobs to do while waiting for your lawyer to arrange your parole, while at the same time, still not giving the prisoners an instant get out of jail free card from the very start.

### CODE REVIEW

**Comment by Bawhoppen** [comment]
Soliciting any and all suggestions. Please flood me with them!

**Comment by lolman360** [comment]
maybe a few windows in the gym area too? so u can contemplate space and stuff while pumping iron

**Comment by Dennok** [comment]
And some windows in dining room too.

And maybe complete grille in space.

**Comment by Archanial** [comment]
Add some radstorm protection somewhere.

**Comment by XDTM** [comment]
Are the cameras avaliable in a screen in the warden's room?

**Comment by GuillaumePrata** [comment]
Would be good to recolor some door, since all are Security airlocks its harder to see at a glance which areas the prisoners have access and which they dont.
How is the camera coverage for the AI/Warden? If the prisoners went around breaking all cameras which areas the AI could still look into? It would be good to have a camera on the safe side of the Visitation just so even if a riot breaks we could have that area for possible peace talks, also a AI holopad there so the AI can talk with prisoners.

**Comment by AtlasRules** [comment]
damn i wanna be a prisoner now

**Comment by Michael-Ashfield** [comment]
We may need to reconsider access on the doors leading from brig to perma. If there is a vistiation room, it makes sense that lawyers should be able to access it to speak with their clients, and maybe detectives if trying to uncover details about a crime.

**Comment by jdawg1290** [comment]
Should we really be giving prisoners a shot at free cuffs?

**Comment by MMMiracles** [comment]
This is honestly a pretty cool revamp of permabrig and I'm 100% gonna adapt the idea for donut.

A few minor things like a few pipes going by through walls when there's a door/window nearby and the upper glass airlock missing an air pipe under it but otherwise it looks pretty good, would reenact Hard Time in.

File: code/game/objects/effects/spawners/lootdrop.dm
Comment (ATH1909) [inline code/game/objects/effects/spawners/lootdrop.dm]: make this come in a syringe, not a bottle

**Comment by ATH1909** [comment]
> Should we really be giving prisoners a shot at free cuffs?

They can already make cablecuffs on their own anyway; it'll be fine.

**Comment by ATH1909** [comment]
https://www.youtube.com/watch?v=Gcv7LGUvBzk

**Comment by ArcaneMusic** [comment]
Alright, so some semantics and opinions from when it came up last night on the discord:

- The hallway below the perma brig is now really, REALLY long. I'd suggest moving around the Re-education chamber to partway into the hallway to save on space.
- Alternatively, consider tightening up the central buildings and rooms so that the hallway around the permabrig is a full circle, so that it's easier for officers to do rounds to check on the prison.
- I'd personally roll both maintenance closets into one single, larger room, just for uniformity, and so that there's one less room that security has to check when looking for a prisoner.

Past that, I like do like how it feels like a more 'full' facility, but it will make it harder for security to maintain without a full team, which is difficult enough on servers, but the QOL changes is well worth it.

**Comment by itseasytosee** [comment]
What if there were a few buzzers around letting you send a robotic message over sec coms?
IE plates are done or I need food clothing needs to be washes ect

**Comment by itseasytosee** [comment]
There may or may not be a few creative ways out from what I can tell.

**Comment by Bawhoppen** [comment]
@lolman360 @Dennok 
I considered it, but I kinda like the seclusion element somewhat. May still add them at some point.
@Archanial 
Cells & Cleaning Closet have Rad Protection.
@XDTM I just copy-pasted them, so they should all have the prison network.
@GuillaumePrata 
Will play around with some door colors. Some are currently glass & some are solid. There isn't camera access into the cells, showers, or cleaning closet. Will add a cam & holopad to southside of the visitation. 
@Michael-Ashfield 
I already thought about that, and I likely will change it. Just need to move the double doors over to the left a little bit.
@jdawg1290 I don't think it's that big of a deal. They're uncommon and people still need to be stunned for them to be useful. It'll be interesting to watch I think.
@MMMiracles I tried to do my best to avoid going into walls, but jeez they're a real bother and I just kinda did a half-hearted job at the end.
@ArcaneMusic Ah yeah, the hallway looked really long to me too, I just didn't want to redesign the execution room. I'll fix it somehow. Also, I think the officers only having entry/patrol points on the south is a perfectly adequate point of fallibility in my eyes. A full observation loop would be too much. Also, one of the two maints is a cleaning closet which prisoners have access to, and the other is one they don't which has the APC. I kinda like this, since the maint won't be used often, but when it does, it would be exciting to have it be utilized. Anyways, I think managing it should be balanced enough, as well. Might create a fun dynamic hopefully.
@itseasytosee The buzzers would be an interesting idea, however I think I'll probably pass on them for now. I'd also be curious to hear what you think the weakpoint are? Unless you mean the intended stuff that requires other player interaction, like hiding in an outbound laundry cart.

**Comment by cacogen** [comment]
yeah i mean i like it

**Comment by itseasytosee** [comment]
Windows are always a weak point I'm not saying remove them but it's not that hard to make a spear if you know what you are doing and how to use your environment to your advantage.
Also In the future, if this gets merged I would be glad to try to code in some buzzers.

**Comment by Farquaar** [comment]
This is really cool, and it's clear you put a lot of thought and work into it. Don't be afraid to add additional security features though. It looks like a prison riot waiting to happen, and that risk might incentivize security officers to execute rather than perma serious offenders, which would be a shame.

**Comment by ATH1909** [comment]
I like how the spare gas masks and oxygen tanks for the prisoners have been locked in a room that they don't have access to (the APC room). Actually, I think that locking life-critical supplies for prisoners behind a maintenance door is something that NT would totally do, so keep that in the game/this PR, please (the roundstart prisoners get internal boxes anyway, they'll be fiiiiine).

File: code/game/objects/effects/spawners/lootdrop.dm
Comment (ATH1909) [inline code/game/objects/effects/spawners/lootdrop.dm]: Maybe we could add screwdrivers to this list too? I like the idea of prisoners being able to make grenades and reconstruct destroyed machines.

Glowsticks might also be a nice thing to add to the table.

If you wanted to, you could add wirecutters to this pool as well, making the screwdriver+crowbar+wirecutters prisoner dream theoretically possible, if incredibly unlikely.

Comment (ATH1909) [reply code/game/objects/effects/spawners/lootdrop.dm]: also, it looks like you've weighted cigarettes a bit too heavily, considering that (unless you have the smoker quirk) a lighter or match for them might not even spawn in the permabrig

**Comment by cacogen** [comment]
mapdiffbot is doing something strange

File: code/game/objects/effects/spawners/lootdrop.dm
Comment (Bawhoppen) [reply code/game/objects/effects/spawners/lootdrop.dm]: I'll consider the other tools, but I am kinda fine with just crowbars being a possibility for now. Also, the logic behind the cigarettes was that you really only needed one lighter to light them, and there's a good few chances for one to spawn. I'll look at it again.

**Comment by Bawhoppen** [comment]
@cacogen It may be related to how I fixed merge conflicts. I will be working on the map in the next few days, and that should hopefully fix it.

File: code/game/objects/effects/spawners/lootdrop.dm
Comment (ATH1909) [reply code/game/objects/effects/spawners/lootdrop.dm]: @Bawhoppen any comment on this?

Comment (Bawhoppen) [reply code/game/objects/effects/spawners/lootdrop.dm]: Oh yes, sorry for the late reply. The original thought for this and also the lighters was to add some potential scarcity, so prisoners might need to trade amongst themselves. Although, it probably would just be better to put it in a syringe directly like you suggest,

**Comment by Bawhoppen** [comment]
Alright, Draft 2: 
image
Made the entrance to the prison wing more public access, so lawyers and detectives may enter visitation, assuming there is an officer to lift the shutters. Also, the execution center has been moved. 
Unfortunately I'm having some trouble with mapmerge and Git, so I can't commit this yet. If anyone wants to tell me any tricks to just start fresh with a new map file that I can commit, please let me know!

**Comment by Dennok** [comment]
> Unfortunately I'm having some trouble with mapmerge and Git, so I can't commit this yet. If anyone wants to tell me any tricks to just start fresh with a new map file that I can commit, please let me know!

To get fresh map just swich git to `master` branch. Then you can use first part of mapmerge. Then swich to branch with your modded map and use second part of mapmerge.

To make all this as one commit you can use `git reset HEAD~[number of commits you make]`
Then commit all changes as once, and `git push --force` it.

Rebase can be used too but i see it more complex.

**Comment by Bawhoppen** [comment]
Ah okay, thanks @Dennok! I tried something similar (downloading fresh map and deleting the .before, then running prepare maps, making the changes, then mapmerging), but it didn't work. Must be because of some specific way it recognizes files or something like that. Anyways, will try this! Thanks again!

**Comment by Bawhoppen** [comment]
well, whatever I did must've not worked

**Comment by Bawhoppen** [comment]
Those commands didn't work for me... but I probably somehow used them wrong though

**Comment by ATH1909** [comment]
What seeds will prisoners have access to in this redesigned permabrig?

**Comment by Archanial** [comment]
Add potted plant somewhere

**Comment by Bawhoppen** [comment]
@ATH1909 Oh yes, I wanted people's inputs on what seeds they'd be best off to have. I currently gave them pumpkin, wheat, ambrosia vulgaris, grass, carrot, tomato, potato, garlic, and onion. But I'm open to changing that totally.
@Archanial Why would there be potted plants in a prison? I could in the visitor waiting area I guess, though Box is quite light on potted plants as is, and I wanna stay in theme.

Also, it seems my error for committing is somehow related to map dictionary keys? Anyone have any ideas? This is the specific error I get when I commit: 
image
My Python should be up to date. Don't have any trouble mapmerging initially for new things, just subsequently.

**Comment by ExcessiveUseOfCobblestone** [comment]
license plates made from mats that can be sold sounds quite based.

**Comment by actioninja** [comment]
Holy fuck

**Comment by Bawhoppen** [comment]
@actioninja Yeah, I considered the size alot... and while this does seem big on the surface, there's lot of walls an individual rooms. I think it will be just about the right size in actual play.

**Comment by Bawhoppen** [comment]
Alright, I think I've finally figured out how to get the technical stuff working again with a few well-placed hard resets and force pushes. 

Anyways, I've been hearing some convincing proposals that the normal brig cells should be merged with perma so that way all the prisoners interact, and I have to say I've been somewhat swayed. Of course, the prisoners would still be held on timer, but we could add an additional name field for timer devices which the warden then manages. This would make for an interesting dynamic, since this would allow contraband to naturally flow in and out, allow for interesting escape dynamics, interactions between random crew and grizzled criminals.
It would also have some disadvantages, like that wardens would have to strip and suit up each and every prisoner, which would get tedious. That could possibly be mitigated by an auto clothes-swapping device of some kind in the brig. This of course would require quite a bit of mapping, as well as some attention to detail for a careful balance that ensures the true perma prisoners can't easily cheese an escape just by making a break for it.

Anyone have any thoughts on this?

**Comment by Bawhoppen** [comment]
Added a normal glass shiv made from cloth and glass shards, alongside the laundry cart.
Sprite credit to EOBGames and JustRandomGuy

Need to fix the inhands file though still and conflicts

**Comment by Bawhoppen** [comment]
Ah jeez... map troubles again. I haven't forgotten about this... just need to spend time figuring out how to fix it once more.

**Comment by ATH1909** [comment]
> Alright, I think I've finally figured out how to get the technical stuff working again with a few well-placed hard resets and force pushes.
> 
> Anyways, I've been hearing some convincing proposals that the normal brig cells should be merged with perma so that all the prisoners would then get to interact, and I have to say, I've been somewhat swayed. Of course, the prisoners would still be held on timer, but we could add an additional name field for timer devices which the warden then manages. This would make for an interesting dynamic, since it would allow contraband to naturally flow in and out, allow for interesting escape dynamics, as well interactions between random crew locked up for graffiti, and grizzled criminals.
> It would also have some disadvantages, like that wardens would have to strip and suit up each and every prisoner, which would get tedious. That could possibly be mitigated by an auto clothes-swapping device of some kind in the brig. This of course would require quite a bit of mapping, as well as some attention to detail for a careful balance that ensures the true perma prisoners can't easily cheese an escape just by making a break for it.
> 
> Anyone have any thoughts on this?

Eh, that sounds like it's gonna be slightly too complex for the average warden/sec officer to handle. They already struggle with remembering to set cell timers; don't make forgetting to set a cell timer permabrig someone with no chance of calling for help to someone through a window or something.

**Comment by BadSS13Player** [comment]
Needs a Prison Architect ghost role.

**Comment by Bawhoppen** [comment]
@ATH1909 Yeah, I haven't given up hope on it as an idea yet, but it's not for this PR either way.
@BadSS13Player Warden console that controls a building drone some day in the future....

**Comment by Bawhoppen** [comment]
Maintainers won't accept deleting and re-adding the map because of the unreadable diff changes, right?

**Comment by Bawhoppen** [comment]
Alright, I finally have mapmerge under control I think, thanks to the help of Spacemaniac. 
I will now address other stuff, and move onto the other maps.

**Comment by Bawhoppen** [comment]
Replaced the drug bottles with pre-filled syringes. Added a few extra drugs to the list as well for them, although I don't do drugs so I have no idea if they're too powerful. Please feel free to let me know!

File: code/game/area/Space_Station_13_areas.dm
Comment (kriskog) [inline code/game/area/Space_Station_13_areas.dm]: The other maps still use this area, leave it in to fix most of the travis errors.

File: code/game/objects/items/kitchen.dm
Comment (kriskog) [inline code/game/objects/items/kitchen.dm]: You need to replace all the old path with the new path on snowcabin.dmm. Its failing travis.

File: code/modules/reagents/reagent_containers/syringes.dm
Comment (kriskog) [inline code/modules/reagents/reagent_containers/syringes.dm]: missing newline

**Review by kriskog — CHANGES_REQUESTED** [review]
These things are making travis angry

File: code/game/area/Space_Station_13_areas.dm
Comment (Bawhoppen) [reply code/game/area/Space_Station_13_areas.dm]: I am planning on editing the other maps as well

File: code/game/objects/items/kitchen.dm
Comment (Bawhoppen) [reply code/game/objects/items/kitchen.dm]: Thanks for the catch

File: code/game/area/Space_Station_13_areas.dm
Comment (kriskog) [reply code/game/area/Space_Station_13_areas.dm]: You arent doing it in this PR, so leave it in because it will fuck up the other maps.

Comment (Bawhoppen) [reply code/game/area/Space_Station_13_areas.dm]: I was gonna do it in this PR though, I just am being slow about it

**Comment by Bawhoppen** [comment]
Since I'll be doing the other maps in a separate PR now, the things I need to finish before this PR can be merged/testmerged are:
-Floyd's License Plate PR to be merged
-Cafeteria things to be sprited
-Add job_changes.dm for Box upping prisoner limit to 4
-Double-check mapping stuff like firelocks and atmos

File: code/modules/reagents/reagent_containers/syringes.dm
Comment (ATH1909) [inline code/modules/reagents/reagent_containers/syringes.dm]: This is enough meth to cause a significant explosion (or two if you split it, I think) if heated, which could be used to break walls down into girders. If that's okay with you (since whoever heats the meth up will take damage from the explosion too and it's a creative way to break out of prison), that's fine, but I figured that you should be made aware of this.

File: code/modules/reagents/reagent_containers/syringes.dm
Comment (ATH1909) [inline code/modules/reagents/reagent_containers/syringes.dm]: Would it be possible to make subtypes of this contain a random volume of chems, with some extra weighting for "0", for that real "spent prison syringe" feel?

Also, consider adding a chance for these to come with labels that may or may not be accurate (70% chance to be accurate, perhaps?).

File: code/modules/reagents/reagent_containers/syringes.dm
Comment (Bawhoppen) [reply code/modules/reagents/reagent_containers/syringes.dm]: Jeez, is that how meth works in real life? Either way, I'm fine with that honestly. Emergent gameplay. If it becomes a problem, we can change it later.

File: code/modules/reagents/reagent_containers/syringes.dm
Comment (Bawhoppen) [reply code/modules/reagents/reagent_containers/syringes.dm]: I kinda like the idea, although I'm not sure I'll get around to it on this first PR.

Comment (kriskog) [reply code/modules/reagents/reagent_containers/syringes.dm]: Do it yourself.

**Comment by actioninja** [comment]
I'm ok with this, fix the issue with ids and it looks ready to merge if you have everything ready.

**Comment by Bawhoppen** [comment]
@actioninja Sorry, was a little busy for a few days. It should all be ready now. I've done some general testing locally, although haven't yet had it test-merged on the actual server. Should we wait for that atleast once, or just jump right in with a full-merge?

**Review by kriskog — CHANGES_REQUESTED** [review]
overlapping pipes
image

Missing pipes
image

stray pipe
image

rest of the structural stuff seems fine

File: code/modules/jobs/map_changes/map_changes.dm
Comment (kriskog) [inline code/modules/jobs/map_changes/map_changes.dm]: You are #undefing twice. Remove the top one

**Review by kriskog — CHANGES_REQUESTED** [review]

**Comment by NalyKalZul** [comment]
i love the look for it, but being from citadel and a fervent botanist.
i can't help but wonder, why is the botany section so well hidden.
and why does it have a bio-generator, unless you filter it with a custom list.
that is going to source of rng mutations alone, it is super high rng but botany gets silly things.
not to mention giving prisoners bags and new unrestrained clothing.

**Comment by Bawhoppen** [comment]
@NalyKalZul Well the other maps had the biogen, so I just carried it over. Also, it's not that hidden right? It's just in its own room.

**Comment by NalyKalZul** [comment]
probably is a citadel thing, i did spend a few greenshift willingly in perma, teaching some sec what horror botany can make if limited on resources, but just given time and left unhanded mostly.
its mostly oldschool botanist that can go ham with left4zed, resulting in ied fruits, or various injecting fruits, trait mutations are a thing if you know what to look for to speed it up.

but you can take away a other item instead, the plant scanner so you cannot see the full trait and chemical results without medical access. i dont think prisoners need to know the exact results of generational plant breeding.

**Pull request merged**
